implementations/micrometer-registry-statsd/build.gradle

```diff
@@ -10,6 +10,9 @@ dependencies {
         // We do not use HTTP modules; exclude to avoid false positive CVE reports
         exclude module: 'netty-codec-http'
     }
+	// We need make sure netty dns resolution works on all macs
+	implementation 'io.netty:netty-resolver-dns-native-macos:+:osx-x86_64'
+	implementation 'io.netty:netty-resolver-dns-native-macos:+:osx-aarch_64'
```

Comment:
Since this isn't a test dependency like the `netty-transport-native-kqueue` dependency, it's probably more important that the version is aligned with other netty versions. As is, I don't think it necessarily will be aligned if the netty version of other netty dependencies is not the latest version due to the `+`. We haven't explicitly managed the netty version so far since we only transitively depend on it through reactor-netty. Maybe that should change to add these dependencies properly.

Reply:
Are you thinking we should add something like `implementation platform('io.netty:netty-bom:4.1.+')` to `dependencies.gradle` like we have set up for the [reactor-bom](https://github.com/micrometer-metrics/micrometer/blob/main/dependencies.gradle#L101)?

Reply:
The challenge as I see it is there's some level of coordination to keep the netty versions in line with what works with reactor-netty. Though in theory there shouldn't be a problem with using a newer patch version of netty than reactor-netty compiles against. So it's probably fine to just control the major and minor as you suggest.

Reply:
I'm not sure how we let the version defined in the platform BOM be used while specifying the classifier with Gradle, though. The whole `:+:classifier` syntax was used because I didn't know how else to specify a classifier without a version.

Reply:
Looking forward to this pr being merged. 🤞🏻 Just a heads up about

> I'm not sure how we let the version defined in the platform BOM be used while specifying the classifier with Gradle, though.

You can try the more verbose form instead to exclude the version if you did want to exclude it 👍🏻 

```groovy
implementation(group: "io.netty", name: "netty-resolver-dns-native-macos", classifier: "osx-x86_64")
implementation(group: "io.netty", name: "netty-resolver-dns-native-macos", classifier: "osx-aarch_64")
```

but then the `testImplementation` below doesn't so 🤷🏻‍♀️

Reply:
I like that syntax, and I think it helps make it more clear. If either of you don't object, I'll update the test dependencies to use this same format since that `+` is a bit confusing.

I added the `netty-bom` and used the suggested new syntax and things look good.

I took a look at gradles resolution with both `./gradlew -q :micrometer-registry-statsd:dependencies` and `./gradlew -q :micrometer-registry-statsd:dependencyInsight --dependency io.netty:netty-resolver-dns-native-macos --configuration runtimeClasspath` and it looks like the netty versions are still 4.1.77.Final which matches what it was before this change